### PR TITLE
Added script for compiling MOF and Meta MOF files

### PR DESCRIPTION
**Pull request by raandree**
#### Pull Request (PR) description

One of the most important tasks of the DSC build pipeline are the compilation scripts for MOF and Meta MOF. These scripts have been moved from the DscWorkshop to here.

#### This Pull Request (PR) fixes the following issues

NA

#### Task list

- [x] Added an entry to the change log under the Unreleased section of the file CHANGELOG.md.
      Entry should say what was changed and how that affects users (if applicable), and
      reference the issue being resolved (if applicable).
- [ ] Resource documentation added/updated in README.md.
- [ ] Resource parameter descriptions added/updated in README.md, schema.mof and comment-based
      help.
- [ ] Comment-based help added/updated.
- [ ] Localization strings added/updated in all localization files as appropriate.
- [ ] Examples appropriately added/updated.
- [ ] Unit tests added/updated. See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [ ] Integration tests added/updated (where possible). See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [ ] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

### CODE REVIEW

**Comment by gaelcolas** [comment]
I disagree with this approach, as the Root config must be updated when we change the Import-DscResource with the composites.
So RootConfig must be in the Source folder of the control repo.

To discuss, but as-is, not ok :)

**Comment by gaelcolas** [comment]
Technically, for the MetaMof it's less of an issue iirc.

**Comment by raandree** [comment]
I see the point. How about making the modules to import part of the configuration data, something like:

```yaml
DscCompositeResourceModules:
  - CommonTasks
  - SomeOtherModule
```

Then really all the scripts except of config data could go into here.

**Comment by gaelcolas** [comment]
Yes, the right way to solve this is to "dynamically" generate the RootConfiguration script, based on a default and what we need to be "imported" with Import-DscResource.
The things to be imported could be in build.yml.
We should probably have a plan B if the file RootConfiguration.ps1 is found in the source folder, use that instead.

**Comment by raandree** [comment]
Done, modules can be configured in the build.yaml:

```yml
Sampler.DscPipeline:
  DscCompositeResourceModules:
  - CommonTasks
  - SomeOtherModule
```

**Comment by gaelcolas** [comment]
Hey, that looks ok, although I'd have preferred if the compilation script (rootConfig.ps1) was usable "as-is".

Basically, leave it as a script, maybe add a `#<imports>`, and have a function to get the content as string and replace the imports.

But what you have should work, need to check a bit more when on PC.

**Comment by gaelcolas** [comment]
Also best to support module version in the imports, in case it's needed

**Comment by gaelcolas** [comment]
Need to remove the import psdsc module (Don't want it by default).

**Comment by raandree** [comment]
> Need to remove the import psdsc module (Don't want it by default).

You mean this?

```powershell
Import-DscResource -ModuleName PSDesiredStateConfiguration
```

**Comment by gaelcolas** [comment]
Yes

**Comment by raandree** [comment]
Updated and tested inside the DscWorkshop.

**Review by gaelcolas — DISMISSED** [review]
Looks good.
We ought to have some tests soonish now.

**Comment by gaelcolas** [comment]
Ah, last thing, the Changelog could be more descriptive about the changes you just made regarding compilatiton.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Adding pipeline tasks and commands from DSC Workshop.
 - Small changes to support easier deployment for individual environments.
+- Added scripts for compiling MOF and Meta MOF files.
```
Comment (gaelcolas) [inline CHANGELOG.md:12]:
```suggestion
- Added scripts for compiling MOF and Meta MOF files without the need for the `rootConfig.ps1` script. It is now a self-contained task that takes parameters from the `Build.yml`.
```

**Pull request merged**